pkg/apis/eksctl.io/v1alpha5/managed_nodegroup_test.go

```diff
@@ -170,4 +170,57 @@ var _ = Describe("Managed Nodegroup Validation", func() {
 			},
 		}),
 	)
+
+	type updateConfigEntry struct {
+		unavailable           *int
+		unavailablePercentage *int
+		maxSize               *int
+		valid                 bool
+	}
+
+	DescribeTable("UpdateConfig", func(e updateConfigEntry) {
+		mng := &ManagedNodeGroup{
+			NodeGroupBase: &NodeGroupBase{
+				AMIFamily: "AmazonLinux2",
+				ScalingConfig: &ScalingConfig{
+					MaxSize: e.maxSize,
+				},
+			},
+			UpdateConfig: &NodeGroupUpdateConfig{
+				MaxUnavailable:           e.unavailable,
+				MaxUnavailablePercentage: e.unavailablePercentage,
+			},
+		}
+		SetManagedNodeGroupDefaults(mng, &ClusterMeta{Name: "managed-cluster"})
+		err := ValidateManagedNodeGroup(mng, 0)
+		if e.valid {
+			Expect(err).ToNot(HaveOccurred())
+		} else {
+			Expect(err).To(HaveOccurred())
+		}
+	},
+		Entry("max unavailable set", updateConfigEntry{
+			unavailable: aws.Int(1),
+			valid:       true,
+		}),
+		Entry("max unavailable specified in percentage", updateConfigEntry{
+			unavailablePercentage: aws.Int(1),
+			valid:                 true,
+		}),
+		Entry("returns an error if both are set", updateConfigEntry{
```

Comment:
Should there be a test to validate that either unavailable or unavailablePercentage is defined?